lspattern/canvas/compiled.py

```
@@ -359,11 +371,23 @@ def _remap_layer_mappings(next_layer: TemporalLayer, node_map2: Mapping[int, int
 
 
 def _build_merged_coord2node(cgraph: CompiledRHGCanvas, next_layer: TemporalLayer) -> dict[PhysCoordGlobal3D, int]:
-    """Build merged coordinate to node mapping."""
-    return {
-        **cgraph.coord2node,
-        **next_layer.coord2node,
-    }
+    """Build merged coordinate to node mapping without duplicate node ids."""
+
+    merged: dict[PhysCoordGlobal3D, int] = dict(cgraph.coord2node)
+    node2coord = {node_id: coord for coord, node_id in merged.items()}
+
+    for node_id, coord in next_layer.node2coord.items():
+        # Avoid overriding an existing node mapped to the same coordinate with a different id.
+        current_node = merged.get(coord)
+        if current_node is not None and current_node != node_id:
+            continue
+        existing_coord = node2coord.get(node_id)
+        if existing_coord is not None and existing_coord != coord:
+            merged.pop(existing_coord, None)
+        merged[coord] = node_id
+        node2coord[node_id] = coord
```

Is it really safe? I think we should raise an error rather than suppress it.
NOTE: If you take it as a temporary patch, you can merge it without completely solving it.